data/src/main/java/io/gnosis/data/repositories/UnstoppableDomainsRepository.kt

```diff
@@ -0,0 +1,36 @@
+package io.gnosis.data.repositories
+
+import com.unstoppabledomains.config.network.model.Network
+import com.unstoppabledomains.exceptions.ns.NamingServiceException
+import com.unstoppabledomains.resolution.Resolution
+import com.unstoppabledomains.resolution.naming.service.NamingServiceType
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+import java.util.concurrent.Callable
+import java.util.concurrent.FutureTask
+
+class BackgroundTask : Callable<String> {
+    var address = ""
+    var domain: String;
+    private val library: Resolution = Resolution.builder()
+            .infura(NamingServiceType.CNS, asString(getKey("INFURA_API_KEY", ""))
```

Comment:
I am not really sure how to get this key from the system.
It might be the reason why `udtestdev-creek.crypto` doesn't resolve. 
I have tried to use the BuildConfig, however, it seems the INFURA_API_KEY is not added for this package.

Reply:
I hard coded the key for now. But you are right the code above doesn't compile/work. So any Infura key should work?

Reply:
ATM `BuildConfig.INFURA_API_KEY` is not available in the `data` module. Maybe we should change that.

Reply:
I think i know why the domain is not resolving. can you try to resolve brad.crypto? 
I believe the library is connected to the mainnet by default. You can overwrite this line
`.chainId(NamingServiceType.CNS, Network.ROPSTEN)`

I don't think we have any rinkeby contract deployed, but will ask tomorrow morning with the team

Reply:
To get the INFURA_API_KEY from `$basedir/project_keys` Please add the following in `data/build.gradle` in the `defaulConfig` section. Right below `NATIVE_CURRENCY_NAME`:
```
        buildConfigField javaTypes.STRING, "INFURA_API_KEY", asString(getKey("INFURA_API_KEY", ""))
```

Then you can use `BuilConfig.INFURA_API_KEY` whereever you need it.

Reply:
@biafra23  Can you test it out? I checked and was very wrong that we don't have rinkeby contracts. In my mind, I was thinking more of a ropsten than rinkeby. 

The commit 80cee73 resolves rinkeby udtestdev-creek.crypto 
The commit 222f84c resolves mainnet domain such as brad.crypto

Reply:
@biafra23 Are you detecting network id somewhere (mainnet or ropsten)? If yes - how can we access it in the code?

@JohnnyJumper The other way if to detect network on library initialisation by using net_version RPC call: https://eth.wiki/json-rpc/API#net_version

Reply:
> The commit 80cee73 resolves rinkeby udtestdev-creek.crypto
Adding the chainId works. 

We do not distinguish between network ids ATM. We just use a different service URL. Currently we support `mainnet`, `rinkeby` and `xdai`. Do you support xdai? 

You could look for `rinkeby` in `BuildConfig.CLIENT_GATEWAY_URL` and use mainnet in any other case.

Reply:
Since they are all different flavors we could also have an explicit `chainId`.

Reply:
@JohnnyJumper

`BuildConfig.BLOCKCHAIN_NET_URL` contains the Infura URL. Please add this to the data module and use it as provider. Please, detect network id on library init (need to update library version).

cc: @biafra23